File: daySix/python/test_question_three.py
from unittest import TestCase

from question_three import *

class TestQuesionThree(TestCase):

    def test_thatWhen_iEnterAnArray_ofIntegers_allTheZeroElement_areAtTheEndOf_theArray(self):
        array_of_numbers = [5,0,3,0,2,0]
        
        actual_array_of_primes = move_zero_element_to_end_of_array_in(array_of_numbers)
        
        expected_array_of_primes = [5,3,2,0,0,0]
        
        self.assertEqual(expected_array_of_primes, actual_array_of_primes) 

File: daySix/Java/TestQuestionThree.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

public class TestQuestionThree {
    
    @Test
    void TestThatWhenIEnterAnArrayOfIntegersAllTheZeroElementsAreMovedToTheEndOfTheArray(){
        //Given
        int [] arrayOfNumbers = {5,0,3,0,2,0};
        
        //When
        int [] actualSortedArray = QuestionThree.moveZeroElementToEndOfArrayIn(arrayOfNumbers);
        
        int [] expectedSortedArray = {5,3,2,0,0,0};
        
        //Check
        assertArrayEquals(expectedSortedArray, actualSortedArray);
    }
}
